tests/unit/src/test/scala/tests/codeactions/ExtractClassLspSuite.scala

```diff
@@ -0,0 +1,498 @@
+       |  }
+       |}
+       |""".stripMargin,
+    s"""|${ExtractClass.title("B")}""".stripMargin,
+    """|package a
+       |
+       |case class A()
+       |
+       |package b {
+       |
+       |  package c {
+       |    case class C()
+       |  }
+       |}
+       |""".stripMargin,
+    newFile = (
+      "B.scala",
+      s"""|package a
+          |package b
+          |case class B()
+          |""".stripMargin
+    )
+  )
+
+  checkExtractedClass(
+    "extract-class-inner-subpackage",
+    """|package a
+       |
+       |case class A()
+       |
+       |package b {
+       |  case class B()
+       |  package c {
+       |case class <<C>>()
+       |  }
+       |}
+       |""".stripMargin,
+    s"""|${ExtractClass.title("C")}""".stripMargin,
+    """|package a
+       |
+       |case class A()
+       |
+       |package b {
+       |  case class B()
+       |  package c {
+       |
+       |  }
+       |}
+       |""".stripMargin,
+    newFile = (
+      "C.scala",
+      s"""|package a
+          |package b
+          |package c
+          |case class C()
+          |""".stripMargin
+    )
+  )
+
+  checkExtractedClass(
+    "extract-class-inner-package-no-other-packages",
+    """|package a
+       |
+       |case class A()
+       |
+       |package b {
+       |case class <<B>>()
+       |}
+       |
+       |package c {
+       |  case class C()
+       |}
+       |""".stripMargin,
+    s"""|${ExtractClass.title("B")}""".stripMargin,
+    """|package a
+       |
+       |case class A()
+       |
+       |package b {
+       |
+       |}
+       |
+       |package c {
+       |  case class C()
+       |}
+       |""".stripMargin,
+    newFile = (
+      "B.scala",
+      s"""|package a
+          |package b
+          |case class B()
+          |""".stripMargin
+    )
+  )
+
+  def checkExtractedClass(
+      name: TestOptions,
+      input: String,
+      expectedActions: String,
+      expectedCode: String,
+      newFile: (String, String),
+      selectedActionIndex: Int = 0
+  )(implicit loc: Location): Unit = {
+    check(
+      name,
+      input,
+      expectedActions,
+      expectedCode,
+      selectedActionIndex,
+      extraOperations = Some(() => {
+        val (fileName, content) = newFile
+        val absolutePath = workspace.resolve(getPath(fileName))
+        assert(
+          absolutePath.exists,
+          s"File $absolutePath should have been created"
+        )
+        assertNoDiff(absolutePath.readText, content)
+      })
+    )
+  }
+
+  def checkFileRenamed(
+      name: TestOptions,
+      input: String,
+      expectedActions: String,
+      newFileName: String,
+      selectedActionIndex: Int = 0,
+      expectNoDiagnostics: Boolean = true,
+      kind: List[String] = Nil,
+      scalafixConf: String = "",
+      scalacOptions: List[String] = Nil,
+      scalaVersion: String = scalaVersion
+  )(implicit loc: Location): Unit = {
+    val fileName: String = "A.scala"
```

Comment:
Would you be able to reuse  the `check` method here? You should be able modify it by adding an additional parameter.